lib/src/generic/interceptors/csrf_interceptor.dart

```diff
@@ -58,9 +60,18 @@ class CsrfInterceptor extends Interceptor {
   Future<Context> onOutgoing(Provider provider, Context context) async {
     // Inject CSRF token into headers.
     if (context is HttpContext) {
-      if (!context.request.headers.containsKey(_header)) {
+      if (context.request.headers.containsKey(_header)) {
+        if (context.request.headers[_header] == null) {
+          throw new ArgumentError('CSRF header value can not be null');
+        }
+        ;
+      } else {
         context.request.headers[_header] = token;
       }
+
+      if (csrfNotRequired.contains(context.method)) {
+        context.request.headers.remove(_header);
+      }
```

Comment:
```dart
if (context.request.headers.containsKey(_header)) {
  // ...
} else if (csrfRequired.contains(context.method)) {
  // set token
}
```

Also note that I switched to a `csrfRequired` list in this example because otherwise the check would be `!csrfNotRequired.contains(...)` which is a confusing double negative